tools/integration-tests/tester/framework/framework.go

```diff
@@ -1,99 +1,85 @@
 // Package framework provides integration test functionality for GoShimmer with a Docker network.
-// It effectively abstracts away all complexity with discovering peers,
-// waiting for them to autopeer and offers easy access to the peers' web API
-// and logs via Docker.
+// It effectively abstracts away all complexity with creating a custom Docker network per test,
+// discovering peers, waiting for them to autopeer and offers easy access to the peers' web API and logs.
 package framework
 
 import (
-	"fmt"
-	"math/rand"
+	"strings"
+	"sync"
 	"time"
 
 	"github.com/docker/docker/client"
 )
 
-// Framework is a wrapper encapsulating all peers
+var (
+	once        sync.Once
+	instance    *Framework
+	instanceErr error
+)
+
+// Framework is a wrapper that provides the integration testing functionality.
 type Framework struct {
-	peers     []*Peer
-	dockerCli *client.Client
+	tester       *DockerContainer
+	dockerClient *client.Client
 }
 
-// New creates a new instance of Framework, gets all available peers within the Docker network and
-// waits for them to autopeer.
-// Panics if no peer is found.
-func New() *Framework {
-	fmt.Printf("Finding available peers...\n")
+// Instance returns the singleton Framework instance.
+func Instance() (*Framework, error) {
```

Comment:
Do not make `instanceErr` a global variable, instead use named return values for `Instance` for the same effect.